src/lib.rs

```diff
@@ -180,6 +180,55 @@ impl<K: Eq + Clone> Events<K> {
     }
 }
 
+/// Optional timeout.
+///
+/// Note that the maximum timeout is `i32::MAX` milliseconds (about 25 days). Longer
+/// timeouts will be silently clipped to `i32::MAX` milliseconds.
+#[derive(Debug, Clone)]
+pub enum Timeout {
+    /// Timeout after a specific duration.
+    After(Duration),
+    /// Never timeout.
+    Never,
+}
+
+impl Timeout {
+    /// Create a timeout with the specified number of seconds.
+    ///
+    /// See [`Timeout`] for an important note about the maximum timeout.
+    pub fn from_secs(seconds: u32) -> Self {
+        Self::After(Duration::from_secs(seconds as u64))
+    }
+
+    /// Create a timeout with the specified number of milliseconds.
+    ///
+    /// See [`Timeout`] for an important note about the maximum timeout.
+    pub fn from_millis(milliseconds: u32) -> Self {
```

Comment:
I used “millis“ and “secs” instead of “milliseconds“ and “seconds” to match `time::Duration`.